### PR TITLE
Update go.mod to be more conservative.

### PR DESCRIPTION
**Pull request by jkh52**
See discussion at https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/381

What I did:

- match requirements with k/k release-1.24 branch
- for k8s.io/api and friends, take latest 1.24 tag (v0.24.8)
- delete indirect require section (to generate from scratch)
- go mod tidy

### CODE REVIEW

**Comment by tallclair** [comment]
I think we should consider resetting the indirects section. See my comment here: https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/428#issuecomment-1340055713

**Comment by jkh52** [comment]
> I think we should consider resetting the indirects section.

Done. Yikes - pretty big delta.

File: go.mod [hunk 1]
```diff
@@ -5,65 +5,67 @@ go 1.18
...
+	golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8 // indirect
+	golang.org/x/sys v0.1.0 // indirect
+	golang.org/x/term v0.1.0 // indirect
+	golang.org/x/text v0.4.0 // indirect
+	golang.org/x/time v0.0.0-20220210224613-90d013bbcef8 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
-	google.golang.org/genproto v0.0.0-20220812140447-cec7f5303424 // indirect
-	google.golang.org/protobuf v1.28.1 // indirect
+	google.golang.org/genproto v0.0.0-20220107163113-42d7afdf6368 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
```
Comment (tallclair) [inline go.mod:60]: Do we need to rebuild the generated protos if this is changing?

Comment (jkh52) [reply go.mod:60]: When I 'make clean' + 'make gen' in this branch, I get a no change.

**Pull request merged**